packages/lit-html/src/lib/directives/live.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {
+  directive,
+  Directive,
+  AttributePart,
+  noChange,
+  nothing,
+  PartInfo,
+  NODE_PART,
+  EVENT_PART,
+  PROPERTY_PART,
+  BOOLEAN_ATTRIBUTE_PART,
+  ATTRIBUTE_PART,
+} from '../lit-html.js';
+
+class LiveDirective extends Directive {
+  constructor(part: PartInfo) {
+    super();
+    if (part.type === EVENT_PART || part.type === NODE_PART) {
+      throw new Error(
+        'The `live` directive is not allowed on text or event bindings'
+      );
+    }
+    if (part.strings !== undefined) {
+      throw new Error('`live` bindings can only contain a single expression');
+    }
+  }
+
+  render(value: unknown) {
+    return value;
+  }
+
+  update(part: AttributePart, [value]: Parameters<this['render']>) {
+    const element = part.element;
+    const name = part.name;
+
+    // TODO (justinfagnani): This is essentially implementing a getLiveValue()
+    // method for each part type. Should that be moved into the AttributePart
+    // interface?
+    if (part.type === PROPERTY_PART) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      if (value === (element as any)[name]) {
+        return noChange;
+      }
+    } else if (part.type === BOOLEAN_ATTRIBUTE_PART) {
+      if (value == element.hasAttribute(name)) {
+        return noChange;
+      }
+    } else if (part.type === ATTRIBUTE_PART) {
+      if (element.getAttribute(name) === String(value)) {
+        return noChange;
+      }
+    }
+    // Setting the part's value to nothing causes its dirty-check to fail so
+    // that it always sets the value
+    part.__value = nothing;
```

Comment:
@sorvell @kevinpschaaf I haven't figured out a better way to do this.

What if we made it public, but named it `_value` and documented it as not for general use?

Reply:
`unset`?

Reply:
The dirty check in attribute part is confusing. Can we add a comment there?

`!((isPrimitive(v) || v === nothing) && v === this.__value) &&
        v !== noChange`

Is `(isPrimitive(v) || v === nothing)` to prevent non-primitive types from setting to attributes? Why bother?

Reply:
I changed this to set a sentinel owned by live() that will always fail the dirty check.

Reply:
`(isPrimitive(v) || v === nothing)` is to limit the dirty check (`v === this.__value`) to primitives and `nothing`. Otherwise just always set the value.

Reply:
Added comment. I tried distributing the `!` but I though it was harder to understand.